redfish_client/resource.py

```diff
@@ -109,6 +117,45 @@ def execute_action(self, action_name, payload):
             return self._connector.post(action.target, payload=payload)
         raise KeyError("Action with {} does not exist".format(action_name))
 
+    def wait_for_stat(self, stat, expected_value, fail_values=list(), poll_every_num_sec=3,
+                      timeout=30*60):
+        """
+        :param stat: list of keys or single key
+        :param expected_value: expected value
+        :param fail_values: fail values
+        :param poll_every_num_sec: number of requests per second
+        :param timeout: timeout in seconds
+        :return: actual value if expected_value matches else return None
+        Raises:
+            TimedOutError: If timeout is exceeded.
+        """
+
+        if "@odata.id" not in self._content:
+            raise Exception("Element does not have '@odata.id' attribute, cannot wait for a stat "
```

Comment:
Now that we've introduced custom Exceptions I suggest we use custom exception here as well.

Reply:
I think we could use KeyError here, since we are actually looking for a key?

Reply:
In my opinion redfish-client should be only raising custom exceptions that all inherit from base custom exception. This way user can easily tell whether the exception is his syntax error or something "expected" ie. something that happened because of unexpected redfish response.

@tadeboro wdyt?

Reply:
I think having a common base exception makes sense, since this allows end-user to catch all our errors with a single catch while debugging/developing.

In this case, custom exception would be preferable over `KeyError`, since user is not requesting this key explicitly. Getting an `@odata.id` value is just an implementation detail of refresh process.

On the other hand, raising a `KeyError` for invalid names in `stat` parameter would make perfect sense, since this is where the end-user is actually talking about keys.